### ISSUE TITLE
Failed to deserialize collection that contains null value

### ISSUE THREAD

**Issue by altrive** [neuecc/Utf8Json#7]
Deserialize object failed, when following condition met
  1. Collection contains null value
  2. Don't specify exact type to Deserialize method, and fallback resolver for object is used.

Following code  reproduce problems. (Confirmed by v1.0.3.1)
``` CSharp
var item = new string[] { "aaa", null, "bbb" };

//Serialize to JSON string
//  -> ["aaa",null,"bbb"]
var json = JsonSerializer.ToJsonString(item);

//Pattern1: OK
//  When specify explicit type
var result1 = JsonSerializer.Deserialize<string[]>(json);

//Pattern2: NG:
//  When specify dynamic type (or specify object[])
var result2 = JsonSerializer.Deserialize<dynamic>(json);
````

StackTrace
``` 
Unhandled Exception: Utf8Json.JsonParsingException: expected:,, actual:n at:11
  at Utf8Json.JsonReader.ReadIsValueSeparatorWithVerify()
  at Utf8Json.JsonReader.ReadIsEndObjectWithSkipValueSeparator(Int32 & count)
  at Utf8Json.Formatters.PrimitiveObjectFormatter.Deserialize(JsonReader & reader, IJsonFormatterResolver formatterResolver)
  at Utf8Json.Formatters.DynamicObjectTypeFallbackFormatter.Deserialize(JsonReader & reader, IJsonFormatterResolver formatterResolver)
```

**Comment by neuecc** [comment]
thank you , PrimitiveObjectFormatter does not handle null correctly.
I've fixed.

**Comment by neuecc** [comment]
I've released `v1.0.4`.

**Comment by altrive** [comment]
Thanks for quick fix!
I've confirmed problem fixed by `v1.0.4 ` package.
